build_posix/aclocal/options.m4

```diff
@@ -247,6 +247,19 @@ no)	wt_cv_crc32_hardware=no
 	AC_MSG_RESULT(no);;
 esac
 
+AH_TEMPLATE(WT_STANDALONE_BUILD,
+    [Define to 1 to support standalone build.])
+AC_MSG_CHECKING(if --disable-standalone-build option specified)
+AC_ARG_ENABLE(standalone-build,
+       [AS_HELP_STRING([--disable-standalone-build],
+           [Disable standalone build support.])], r=$disableval, r=no)
+case "$r" in
+no)    wt_cv_disable_standalone_build=no
+       AC_DEFINE(WT_STANDALONE_BUILD);;
+*)     wt_cv_disable_standalone_build=yes;;
+esac
+AC_MSG_RESULT($wt_cv_disable_standalone_build)
+
```

Comment:
I ran this script:
```
echo '1: ./configure'
./configure > XX
egrep -i standalone XX wiredtiger_config.h
echo; echo

echo '2: ./configure --disable-standalone-build'
./configure --disable-standalone-build > XX
egrep -i standalone XX wiredtiger_config.h
echo; echo

echo '3: ./configure --disable-standalone-build=no'
./configure --disable-standalone-build=no > XX
egrep -i standalone XX wiredtiger_config.h
echo; echo

echo '4: ./configure --disable-standalone-build=yes'
./configure --disable-standalone-build=yes > XX
egrep -i standalone XX wiredtiger_config.h
echo; echo

echo '5: ./configure --enable-standalone-build'
./configure --enable-standalone-build > XX
egrep -i standalone XX wiredtiger_config.h
echo; echo

echo '6: ./configure --enable-standalone-build=no'
./configure --enable-standalone-build=no > XX
egrep -i standalone XX wiredtiger_config.h
echo; echo

echo '7: ./configure --enable-standalone-build=yes'
./configure --enable-standalone-build=yes > XX
egrep -i standalone XX wiredtiger_config.h
echo; echo
```
and got this output:
```
1: ./configure
XX:checking if --disable-standalone-build option specified... no
wiredtiger_config.h:/* Define to 1 to support standalone build. */
wiredtiger_config.h:#define WT_STANDALONE_BUILD 1


2: ./configure --disable-standalone-build
XX:checking if --disable-standalone-build option specified... yes
wiredtiger_config.h:/* Define to 1 to support standalone build. */
wiredtiger_config.h:/* #undef WT_STANDALONE_BUILD */


3: ./configure --disable-standalone-build=no
configure: error: invalid feature name: standalone-build=no
wiredtiger_config.h:/* Define to 1 to support standalone build. */
wiredtiger_config.h:/* #undef WT_STANDALONE_BUILD */


4: ./configure --disable-standalone-build=yes
configure: error: invalid feature name: standalone-build=yes
wiredtiger_config.h:/* Define to 1 to support standalone build. */
wiredtiger_config.h:/* #undef WT_STANDALONE_BUILD */


5: ./configure --enable-standalone-build
XX:checking if --disable-standalone-build option specified... yes
wiredtiger_config.h:/* Define to 1 to support standalone build. */
wiredtiger_config.h:/* #undef WT_STANDALONE_BUILD */


6: ./configure --enable-standalone-build=no
XX:checking if --disable-standalone-build option specified... yes
wiredtiger_config.h:/* Define to 1 to support standalone build. */
wiredtiger_config.h:/* #undef WT_STANDALONE_BUILD */


7: ./configure --enable-standalone-build=yes
XX:checking if --disable-standalone-build option specified... yes
wiredtiger_config.h:/* Define to 1 to support standalone build. */
wiredtiger_config.h:/* #undef WT_STANDALONE_BUILD */
```
which wasn't what I expected. I didn't dig into it, but I'm not clear on why:

`--disable-standalone-build` doesn't take arguments, but `--enable-standalone-build` does

the #5/7 tests look wrong, --enable-standalone-build doesn't do what I'd expect

I might be misremembering how --enable/--disable work, but my recollection is that any option can be configured with either form.

Anyway, I'm happy to dig into this tomorrow if you want me to propose a change to fix it, or feel free to ignore me if I'm just wrong.1

Reply:
I updated the code and ran the above script. Looks like now the code does as expected.

I tried running other `disable` configurations with arguments but didn't work.
 
```
1: ./configure
XX:checking if --disable-standalone-build option specified... no
wiredtiger_config.h:/* Define to 1 to support standalone build. */
wiredtiger_config.h:#define WT_STANDALONE_BUILD 1


2: ./configure --disable-standalone-build
XX:checking if --disable-standalone-build option specified... yes
wiredtiger_config.h:/* Define to 1 to support standalone build. */
wiredtiger_config.h:/* #undef WT_STANDALONE_BUILD */


3: ./configure --disable-standalone-build=no
configure: error: invalid feature name: standalone-build=no
wiredtiger_config.h:/* Define to 1 to support standalone build. */
wiredtiger_config.h:/* #undef WT_STANDALONE_BUILD */


4: ./configure --disable-standalone-build=yes
configure: error: invalid feature name: standalone-build=yes
wiredtiger_config.h:/* Define to 1 to support standalone build. */
wiredtiger_config.h:/* #undef WT_STANDALONE_BUILD */


5: ./configure --enable-standalone-build
XX:checking if --disable-standalone-build option specified... no
wiredtiger_config.h:/* Define to 1 to support standalone build. */
wiredtiger_config.h:#define WT_STANDALONE_BUILD 1


6: ./configure --enable-standalone-build=no
XX:checking if --disable-standalone-build option specified... yes
wiredtiger_config.h:/* Define to 1 to support standalone build. */
wiredtiger_config.h:/* #undef WT_STANDALONE_BUILD */


7: ./configure --enable-standalone-build=yes
XX:checking if --disable-standalone-build option specified... no
wiredtiger_config.h:/* Define to 1 to support standalone build. */
wiredtiger_config.h:#define WT_STANDALONE_BUILD 1

```